### PR TITLE
Update JuMP0.19

### PR DESCRIPTION
**Pull request by YoungFaithful**
Update to JuMP 0.19 done

### CODE REVIEW

File: REQUIRE [hunk 1]
```diff
@@ -3,4 +3,4 @@ Reexport
 StatsKit
 JLD2
 FileIO
-JuMP 0.18 0.19
+JuMP
```
Comment (YoungFaithful) [inline REQUIRE]: No 0.18 requirement

File: examples/workflow_example_cep.jl [hunk 2]
```diff
@@ -17,30 +17,30 @@ ts_full_data = run_clust(ts_input_data;method="kmeans",representation="centroid"
 
 ## OPTIMIZATION EXAMPLES##
 # select solver
-solver=ClpSolver()
+optimizer=Clp.Optimizer
```
Comment (YoungFaithful) [inline examples/workflow_example_cep.jl]: Different introduction of Optimizer

File: src/optim_problems/opt_cep.jl [hunk 3]
```diff
@@ -52,11 +52,12 @@ setting up the basic core elements for a CEP-model
 function setup_opt_cep_basic(ts_data::ClustData,
                             opt_data::OptDataCEP,
                             opt_config::Dict{String,Any},
-                            solver::Any;
+                            optimizer::DataType,
+                            optimizer_config::Dict{Symbol,Any};
```
Comment (YoungFaithful) [inline src/optim_problems/opt_cep.jl]: optimizer config allows you to use specific keywords for the specific solver

File: src/optim_problems/opt_cep.jl [hunk 4]
```diff
@@ -52,11 +52,12 @@ setting up the basic core elements for a CEP-model
...
                             kwargs...)
    ## MODEL CEP ##
    # Initialize model
-   model=JuMP.Model(solver=solver)
+   model =  JuMP.Model(with_optimizer(optimizer;optimizer_config...))
```
Comment (YoungFaithful) [inline src/optim_problems/opt_cep.jl]: optimizer and optimizer_config is passed on to JuMP

File: src/optim_problems/opt_cep.jl [hunk 5]
```diff
@@ -147,7 +152,7 @@ function setup_opt_cep_fix_design_variables!(cep::OptModelCEP,
 
   ## VARIABLES ##
   # Capacity
-  push!(cep.info,"CAP[tech, 'ex', node] = existing infrastructure ∀ node, tech")
+  push!(cep.info,"CAP[tech, 'new', node] = existing infrastructure ∀ node, tech")
   @constraint(cep.model, [node=set["nodes"], tech=set["tech"]], cep.model[:CAP][tech,"new",node]==get_cep_variable_value(cap,[tech, "new", node]))
```
Comment (YoungFaithful) [inline src/optim_problems/opt_cep.jl]: Small change in caption

File: src/optim_problems/opt_cep.jl [hunk 6]
```diff
@@ -82,7 +83,11 @@ function setup_opt_cep_basic_variables!(cep::OptModelCEP,
   @variable(cep.model, COST[account=set["account"],impact=set["impact"],tech=set["tech"]])
   # Capacity
   push!(cep.info,"Variable CAP[tech, infrastruct, nodes] ≥ 0 in MW]")
+  if "tech_transmission" in keys(set) #if transmission is in tech, exclude the transmission technology from the setup of variable CAP
```
Comment (YoungFaithful) [inline src/optim_problems/opt_cep.jl]: For case of transmission being in tech of data, it is more elegant to not setup CAP["trans"]=0, but not set it up entirely

File: src/optim_problems/opt_cep.jl [hunk 7]
```diff
@@ -524,8 +526,9 @@ function solve_opt_cep(cep::OptModelCEP,
                             ts_data::ClustData,
                             opt_data::OptDataCEP,
                             opt_config::Dict{String,Any})
-  status=solve(cep.model)
-  objective=getobjectivevalue(cep.model)
+  optimize!(cep.model)
+  status=Symbol(termination_status(cep.model))
+  objective=objective_value(cep.model)
```
Comment (YoungFaithful) [inline src/optim_problems/opt_cep.jl]: Different way to call optimize

File: src/utils/datastructs.jl [hunk 8]
```diff
@@ -109,18 +109,19 @@ Constructor for OptVariable taking JuMP Array and type (ov-operational variable
...
+  for axe in jumparray.axes
     for (name, val) in cep.set
       if axe==val
         push!(axes_names, name)
         break
       end
     end
   end
-  OptVariable(jumparray.innerArray,axes_names,jumparray.indexsets,type)
+  OptVariable(round.(jumparray.data;digits=round_digits),axes_names,jumparray.axes,type)
```
Comment (YoungFaithful) [inline src/utils/datastructs.jl]: Opt Variables can immediately be rounded to significant digits

File: src/optim_problems/run_opt.jl [hunk 9]
```diff
@@ -59,8 +59,8 @@ what you can add to the opt_config:
 function run_opt(ts_data::ClustData,
                     opt_data::OptDataCEP,
                     opt_config::Dict{String,Any},
-                    fixed_design_variables::Dict{String,OptVariable};
-                    solver::Any=CbcSolver(),
+                    fixed_design_variables::Dict{String,OptVariable},
+                    optimizer::DataType;
```
Comment (YoungFaithful) [inline src/optim_problems/run_opt.jl]: an optimizer is required now

**Comment by holgerteichgraeber** [comment]
I added one commit to make exact k-medoids work with Gurobi in 0.19, and updated an example file.

File: src/optim_problems/run_opt.jl (hunk 9 above)
Comment (YoungFaithful) [reply src/optim_problems/run_opt.jl]: we would need to be dependent on package Cbc then and include `using Cbc` I don't think that's so elegant

File: src/optim_problems/opt_cep.jl (hunk 5 above)
Comment (YoungFaithful) [reply src/optim_problems/opt_cep.jl]: Mistake in the former caption

File: src/optim_problems/run_opt.jl [hunk 10]
```diff
@@ -4,13 +4,13 @@ organizing the actual setup and run of the CEP-Problem
 """
 function run_opt(ts_data::ClustData,
                     opt_data::OptDataCEP,
-                    opt_config::Dict{String,Any};
-                    solver::Any=CbcSolver(),
+                    opt_config::Dict{String,Any},
+                    optimizer::DataType;
```
[the remark this answers is not in the document]
Comment (YoungFaithful) [reply src/optim_problems/run_opt.jl]: No, I'm not sure

File: src/optim_problems/run_opt.jl (hunk 9 above)
Comment (YoungFaithful) [reply src/optim_problems/run_opt.jl]: We would then be dependent on another package by `using Cbc` I think that's not so elegant

Comment (holgerteichgraeber) [reply src/optim_problems/run_opt.jl]: Yep, agreed, that makes sense. That was a nice feature in the old JuMP that doesn't work anymore with optimizer I guess...

**Comment by holgerteichgraeber** [comment]
Thank you for the great commenting on this PR by the way, really helped me go through easily.

**Pull request merged**